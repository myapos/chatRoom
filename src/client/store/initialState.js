const initialState = {
  'testState': 'test',
  received: [],
  entered: false,
  username: '',
  lastname: '',
};

export default initialState;
