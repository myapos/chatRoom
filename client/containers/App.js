import React from 'react';
import { connect } from 'react-redux';
import openSocket from 'socket.io-client';
import Display from '../components/Display';
import Entry from '../components/Entry';
import Enter from '../components/Enter';
import * as actions from '../store/actions';

import '../css/css.styl';
import '../css/fontawesome-free-5.7.2-web/css/all.css';
import { SERVER_PORT } from '../../constants/common';

const socket = openSocket(`http://localhost:${SERVER_PORT}`);

// https://medium.com/dailyjs/combining-react-with-socket-io-for-real-time-goodness-d26168429a34

const App = (props) =>
  <div className="wrapper">
    {
      props.entered ?
        <React.Fragment>
          < Display socket={socket} />
          <Entry socket={socket} />
        </React.Fragment>
        : <Enter />
    }
  </div>

export default connect(state => state, actions)(App);