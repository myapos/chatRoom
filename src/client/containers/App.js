import React, { Component } from 'react';
import { connect } from 'react-redux';
import openSocket from 'socket.io-client';
import PropTypes from 'prop-types';
import Display from '../components/Display';
import Entry from '../components/Entry';
import Enter from '../components/Enter';
import WhoIsTyping from '../components/WhoIsTyping';
import ListOfUsers from '../components/ListOfUsers';
import * as actions from '../store/actions';

import '../css/css.styl';
import '../css/fontawesome-free-5.7.2-web/css/all.css';
import { SERVER_PORT } from '../../../constants/common';

console.log('SERVER_PORT from client', SERVER_PORT);

let socket = {};
if (process.env.NODE_ENV === 'development') {
  socket = openSocket(`http://localhost:${SERVER_PORT}`);
} else {
  socket = openSocket('https://chatroomsimpledemo.herokuapp.com');
}

// https://medium.com/dailyjs/combining-react-with-socket-io-for-real-time-goodness-d26168429a34

class App extends Component {
  componentDidMount () {
    console.log('');
    const { resetIdleTickTimer, whoIsTyping } = this.props;
    typeof document !== 'undefined'
      && document.body.addEventListener('mouseover', () => {
        resetIdleTickTimer();
      });
    if (!socket._callbacks['$is typing'] || socket._callbacks['$is typing'].length < 1) {
      socket.on('is typing', who => {
        whoIsTyping(who);
      });
    }
  }

  render () {
    const { entered, resetIdleTickTimer } = this.props;

    // get value from storage

    const storedEnter = localStorage.getItem('entered');

    return (
      <div className="bigWrapper" >
        <div
          onClick={() => resetIdleTickTimer()}
          onFocus={() => resetIdleTickTimer()}
          className="wrapper" >
          {
            entered || storedEnter === 'true'
              ? <React.Fragment>
                <Display socket={socket} />
                <WhoIsTyping />
                <Entry socket={socket} />
              </React.Fragment>
              : <Enter />
          }
        </div>
        {
          entered || storedEnter === 'true'
            ? <ListOfUsers />
            : null
        }

      </div>
    );
  }
}

App.propTypes = {
  entered: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  resetIdleTickTimer: PropTypes.func,
  whoIsTyping: PropTypes.func,
};
export default connect(state => state, actions)(App);
