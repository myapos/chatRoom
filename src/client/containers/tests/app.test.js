import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import configureStore from '../../store/store';
import App from '../App';
import initialState from '../../store/initialState';
import setLocalStorage from '../../utils/tests/setLocalStorage';

jest.setTimeout(10000);
// https://github.com/facebook/jest/issues/1456
describe('App testing', () => {
  beforeAll(() => {
    // mock - redefine localStorage for node environment
    // https://auth0.com/blog/testing-react-applications-with-jest/
    setLocalStorage();
  });

  test('not entered', async () => {
    // mount(<Enter props={props} />);
    const tree = await renderer
      .create(
        <Provider store={configureStore(initialState)}>
          <App />
        </Provider>
      )
      .toJSON();

    await expect(tree).toMatchSnapshot();
    // done();
  });
});
