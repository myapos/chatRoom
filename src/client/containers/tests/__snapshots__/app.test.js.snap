// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App testing not entered 1`] = `
<div
  className="wrapper"
  onClick={[Function]}
  onFocus={[Function]}
>
  <div
    className="enter"
  >
    <div
      className="introText"
    >
       Please enter your Name and hit enter!
    </div>
    <div
      className="inputs"
    >
      <input
        className="holder"
        onKeyPress={[Function]}
        placeholder="name"
        type="text"
      />
      <input
        className="holder"
        onKeyPress={[Function]}
        placeholder="lastname"
        type="text"
      />
    </div>
    <div
      className="myBtn clearFloat"
      onClick={[Function]}
    >
      Enter
    </div>
  </div>
</div>
`;
